File: data/src/main/java/org/oppia/data/backends/gae/model/Subtopic.java
package org.oppia.data.backends.gae.model;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

import java.lang.reflect.Type;
import java.util.List;

@AutoValue
public abstract class Subtopic {

    /*
    * Parent Hierarchy: Subtopic
    */

    //TODO: AutoJSON null data error

    @SerializedName("topic_id") public abstract String getTopicId();
    @SerializedName("subtopic_id") public abstract String getSubtopicId();
    @SerializedName("page_contents") public abstract SubtopicPageContents getPageContents();
    
    public static TypeAdapter<Subtopic> createTypeAdapter(Gson gson) {
        return new AutoValue_Subtopic.GsonTypeAdapter(gson);
    }

}

File: data/src/main/java/org/oppia/data/backends/gae/model/Subtopic.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Subtopic(

  /*
   * Ignore:
   * is_moderator, is_admin, username, user_email, iframed, is_topic_manager, is_super_admin,
   * additional_angular_modules
   */

  //@Json(name = "topic_id") val topicId: String,
  //@Json(name = "subtopic_id") val subtopicId: String,
  @Json(name = "subtopic_title") val subtopicTitle: String,
  @Json(name = "page_contents") val pageContents: SubtopicPageContents

)
